Packet++/src/Packet.cpp

```diff
@@ -609,6 +609,9 @@ namespace pcpp
 			// assuming header length of the layer that requested to be extended hasn't been enlarged yet
 			size_t headerLen = curLayer->getHeaderLen() + (curLayer == layer ? numOfBytesToExtend : 0);
 			dataPtr += headerLen;
+
+			if (dataPtr > m_RawPacket->getRawData() + m_RawPacket->getRawDataLen())
+				break;
```

Comment:
How can such a thing happen? `curLayer->getHeaderLen()` should never exceed the packet data, unless we have a bug in one of the layers